devtools/cli/src/main/java/io/quarkus/cli/plugin/Plugin.java

```diff
@@ -0,0 +1,100 @@
+package io.quarkus.cli.plugin;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+import io.quarkus.runtime.annotations.RegisterForReflection;
+
+@JsonInclude(Include.NON_NULL)
+@RegisterForReflection
+public class Plugin {
+
+    private final String name;
+    private final PluginType type;
+    private final Optional<String> location;
+    private final Optional<String> description;
+
+    /**
+     * Autoremovalbe are the plugins that are meant to be removed when no longer available.
+     * Examples:
+     * - a binary no longer found in path.
+     * - an extension provided plugin of a removed extension.
+     * - a jbang alias no longer available
+     * Examples of plugin that should not be considered autoremovable:
+     * - A url pointing to a jar
+     * - An GACTV
+     *
+     * Remote plugins are expensive to check for availability and thus we avoid to treat them
+     * as autoRemovable
+     */
+    private final boolean autoRemovable;
+
+    /**
+     * This is mostly used for testing.
+     */
+    @JsonIgnore
+    private final Optional<Path> catalogLocation;
+
+    Plugin() {
+        this(null, PluginType.jbang);
+    }
+
+    public Plugin(String name, PluginType type) {
+        this(name, type, Optional.empty(), Optional.empty(), false, Optional.empty());
+    }
+
+    public Plugin(String name, PluginType type, Optional<String> location, Optional<String> description) {
+        this(name, type, location, description, false, Optional.empty());
+    }
+
+    public Plugin(String name, PluginType type, Optional<String> location, Optional<String> description,
+            boolean autoRemovable) {
+        this(name, type, location, description, false, Optional.empty());
+    }
+
+    public Plugin(String name, PluginType type, Optional<String> location, Optional<String> description, boolean autoRemovable,
+            Optional<Path> catalogLocation) {
+        this.name = name;
+        this.type = type;
+        this.description = description;
```

Comment:
```suggestion
        this.description = description==null?Optional.empty():description;
```

doesn't feel right but at least let me try things out without fail and I get:

```
 qss plugin list --installable
    Name                   	 Type   	 Location                                	 Description
    aggregate-pom-gen      	 jbang  	 aggregate-pom-gen
    sqlsh                  	 jbang  	 sqlsh
    quarkus-kill           	 binary 	 /Users/manderse/.jbang/bin/quarkus-kill
    quarkus-fmt@quarkusio  	 jbang  	 quarkus-fmt@quarkusio
    quarkus-kill@quarkusio 	 jbang  	 quarkus-kill@quarkusio
    ```

Reply:
I think that we don't have much of a choice. Apparently, jackson passes null values to the constructor, therefore we need to use guards.